src/main/java/com/labsynch/labseer/chemclasses/rdkit/BBChemStructureService.java

```diff
@@ -0,0 +1,338 @@
+
+				JsonNode averageMolWeightNode = responseJsonNode.get("molecular_weight");
+				rdkitStructure.setAverageMolWeight(averageMolWeightNode.asDouble());
+
+				JsonNode exactMolWeightNode = responseJsonNode.get("molecular_weight");
+				rdkitStructure.setExactMolWeight(-1.0);
+
+				JsonNode smilesNode = responseJsonNode.get("smiles");
+				rdkitStructure.setSmiles(smilesNode.asText());
+
+				JsonNode molecularFormulaNode = responseJsonNode.get("molecular_formula");
+				rdkitStructure.setMolecularFormula(molecularFormulaNode.asText());
+
+			}
+
+
+		} catch (Exception e) {
+			logger.error("Error posting to preprocessor service: " + e.getMessage());
+			throw new CmpdRegMolFormatException("Error posting to preprocessor service: " + e.getMessage());
+		}
+
+		// Set recorded date todays date. This simplifies the code when we need to persist the structure
+		// in various places.
+		rdkitStructure.setRecordedDate(new Date());
+		return rdkitStructure;
+	}
+
+	@Override
+	public String getSDFFromRDkitStructure(RDKitStructure rdkitStructure) throws IOException{
+		// Read the preprocessor settings as json
+		JsonNode jsonNode = getPreprocessorSettings();
+
+		// Extract the processURL
+		JsonNode urlNode = jsonNode.get("exportSDFURL");
+		if(urlNode == null || urlNode.isNull()) {
+			logger.error("Missing preprocessorSettings exportSDFURL!!");
+		}
+		String url = urlNode.asText();
+
+		// Get the standardization actions and options
+		ObjectMapper mapper = new ObjectMapper();
+		ObjectNode requestData = mapper.createObjectNode();
+
+		// Create sdfs arraynode and mol node
+		ArrayNode sdfsNode = mapper.createArrayNode();
+		ObjectNode molNode = mapper.createObjectNode();
+
+
+		// Add the properties to the mol node
+		ArrayNode propertiesNode = mapper.createArrayNode();
+		for (Map.Entry<String,String> entry : rdkitStructure.getProperties().entrySet()) {
+			ObjectNode propertyNode = mapper.createObjectNode();
+			propertyNode.put("name", entry.getKey());
+			propertyNode.put("value", entry.getValue());
+			propertiesNode.add(propertyNode);
+
+		}
+		molNode.put("properties", propertiesNode);
+
+		// Add the structure to the mol node
+		molNode.put("sdf", rdkitStructure.getMol());
+
+		// Add the mol node to the sdfs array node
+		sdfsNode.add(molNode);
+
+		// Add the sdfs to the request data
+		requestData.put("sdfs", sdfsNode);
+
+		// Post to the service and parse the response
+		String requestString = requestData.toString();
+		logger.info("requestString: " + requestString);
+		String postResponse = SimpleUtil.postRequestToExternalServer(url, requestString, logger);
+		logger.info("Got response: "+ postResponse);
+
+		// Parse the response json
+		ObjectMapper responseMapper = new ObjectMapper();
+		JsonNode responseNode = responseMapper.readTree(postResponse);
+		return responseNode.asText();
+	}
+
+	@Override
+	public List<RDKitStructure> getRDKitStructuresFromSDFService(String molfile) throws CmpdRegMolFormatException {
```

Comment:
"parse" might be a more meaningful word to put into the function name given that this is a proxy around a "parse" endpoint of bbchem